src/styles/main.scss

```diff
@@ -0,0 +1,58 @@
+@import '../styles/variables';
+
+body {
+  margin: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  font-family: $font-family;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: calc(7 * ($day-size + $gap));
+  padding: $padding;
+  &__day {
+    position: relative;
+    width: $day-size;
+    height: $day-size;
+    border: 1px solid black;
+    box-sizing: border-box;
+    background-color: $day-color;
+    transition: all 0.5s ease;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+        font-size: $day-font-size;
+      }
+    }
```

Comment:
It's a good practice to avoid such deeply nested loops inside your SCSS because it can make the code harder to read and maintain. Consider creating a mixin for the repetitive code or using a different approach to apply styles to the elements.